pkg/integrations/cadvisor/cadvisor.go

```diff
@@ -0,0 +1,255 @@
+	}
+	for _, d := range c.DisabledMetrics {
+		if err := c.disabledMetricsSet.Set(d); err != nil {
+			return fmt.Errorf("failed to set disabled metric: %w", err)
+		}
+	}
+
+	for _, e := range c.EnabledMetrics {
+		if err := c.enabledMetricsSet.Set(e); err != nil {
+			return fmt.Errorf("failed to set enabled metric: %w", err)
+		}
+	}
+
+	if len(c.enabledMetricsSet) > 0 {
+		c.includedMetrics = c.enabledMetricsSet
+	} else {
+		c.includedMetrics = container.AllMetrics.Difference(c.disabledMetricsSet)
+	}
+	return err
+}
+
+// Name returns the name of the integration that this config represents.
+func (c *Config) Name() string {
+	return "cadvisor"
+}
+
+// CommonConfig returns the common settings shared across all configs for
+// integrations.
+func (c *Config) CommonConfig() config.Common {
+	return c.Common
+}
+
+// InstanceKey returns the hostname:port of the cadvisor API server.
+func (c *Config) InstanceKey(agentKey string) (string, error) {
+	return agentKey, nil
+}
+
+// NewIntegration creates a new cadvisor integration
+func (c *Config) NewIntegration(logger log.Logger) (integrations.Integration, error) {
+	return New(logger, c)
+}
+
+func init() {
+	integrations.RegisterIntegration(&Config{})
+}
+
+// New creates a new cadvisor integration
+func New(logger log.Logger, c *Config) (integrations.Integration, error) {
+	klog.SetLogger(logger)
+
+	// Do gross global configs. This works, so long as there is only one instance of the cAdvisor integration
+	// per host.
+	// Containerd
+	containerd.ArgContainerdEndpoint = &c.Containerd
+	containerd.ArgContainerdNamespace = &c.ContainerdNamespace
+
+	// Docker
+	docker.ArgDockerEndpoint = &c.Docker
+	docker.ArgDockerTLS = &c.DockerTLS
+	docker.ArgDockerCert = &c.DockerTLSCert
+	docker.ArgDockerKey = &c.DockerTLSKey
+	docker.ArgDockerCA = &c.DockerTLSCA
+
+	// Raw
+	raw.DockerOnly = &c.DockerOnly
+
+	// Only using in-memory storage, with no backup storage for cadvisor stats
+	memoryStorage := memory.New(c.StorageDuration, []storage.StorageDriver{})
+
+	sysFs := sysfs.NewRealSysFs()
+
+	collectorHTTPClient := http.Client{}
+
+	rm, err := manager.New(memoryStorage, sysFs, manager.HousekeepingConfigFlags, c.includedMetrics, &collectorHTTPClient, c.RawCgroupPrefixWhitelist, c.EnvMetadataWhitelist, c.PerfEventsConfig, time.Duration(c.ResctrlInterval))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a manager: %w", err)
+	}
+
+	if err := rm.Start(); err != nil {
+		return nil, fmt.Errorf("failed to start manager: %w", err)
+	}
+
+	containerLabelFunc := metrics.DefaultContainerLabels
+	if !c.StoreContainerLabels {
+		containerLabelFunc = metrics.BaseContainerLabels(c.WhitelistedContainerLabels)
+	}
+
+	goCol := collectors.NewGoCollector()                                         // This is already emitted by the agent, but not with the integration job name.
+	procCol := collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}) // Same as above
```

Comment:
This feels heavy handed. Ultimately the agent emits these metrics on it's own, and should probably own anything that's process/host wide.

That said, the dashboards in the onboarding integrations expect everything to have labels unique to the integration.

Thoughts?

Reply:
@rgeyer where are these dashboards located? I just wanted to take a peek 👀

Reply:
I was curious as to why/how they might be using them so this is an interesting find as we may not need this after all... ?